platforms/jvm/testing-jvm/src/integTest/groovy/org/gradle/testing/nonclassbased/AbstractNonClassBasedTestingIntegrationTest.groovy

```diff
@@ -16,34 +16,40 @@
 
 package org.gradle.testing.nonclassbased
 
+import org.gradle.api.internal.tasks.testing.report.VerifiesGenericTestReportResults
+import org.gradle.api.internal.tasks.testing.report.generic.GenericTestExecutionResult.TestFramework
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
 import testengines.TestEnginesFixture
 
 /**
  * Abstract base class for tests that exercise and demonstrate incorrect Non-Class-Based Testing setups.
  */
-abstract class AbstractNonClassBasedTestingIntegrationTest extends AbstractIntegrationSpec implements TestEnginesFixture {
+abstract class AbstractNonClassBasedTestingIntegrationTest extends AbstractIntegrationSpec implements TestEnginesFixture, VerifiesGenericTestReportResults {
     public static final DEFAULT_DEFINITIONS_LOCATION = "src/test/definitions"
 
-    protected void sourcesPresentAndNoTestsFound() {
-        failureCauseContains("There are test sources present and no filters are applied, but the test task did not discover any tests to execute. This is likely due to a misconfiguration. Please check your test configuration. If this is not a misconfiguration, this error can be disabled by setting the 'failOnNoDiscoveredTests' property to false.")
+    @Override
+    TestFramework getTestFramework() {
+        return TestFramework.JUNIT_JUPITER
```

Comment:
❓ I'm curious why you return this value here. Wouldn't `CUSTOM` be more appropriate?